deepchem/feat/molecule_featurizers/one_hot_featurizer.py

```diff
@@ -42,57 +43,113 @@ def __init__(self, charset: List[str] = ZINC_CHARSET, max_length: int = 100):
 
-  def _featurize(self, mol: RDKitMol) -> np.ndarray:
-    """Compute one-hot featurization of this molecule.
+  def featurize(self, datapoints: Iterable[Any],
+                log_every_n: int = 1000) -> np.ndarray:
+    """Featurize strings or mols.
 
     Parameters
     ----------
-    mol: rdkit.Chem.rdchem.Mol
-      RDKit Mol object
+    datapoints: list
+      A list of either strings or RDKit molecules.
+    log_every_n: int, optional (default 1000)
+      How many elements are featurized every time a featurization is logged.
+    """
+    datapoints = list(datapoints)
+    if (len(datapoints) < 1):
+      return np.array([])
+    # Featurize data using featurize() in grandparent class
+    return Featurizer.featurize(self, datapoints, log_every_n)
+
+  def _featurize(self, datapoint: Any):
+    # Featurize str data
+    if (type(datapoint) == str):
+      return self._featurize_string(datapoint)
+    # Featurize mol data
+    else:
+      return self._featurize_mol(datapoint)
+
+  def _featurize_string(self, string: str) -> np.ndarray:
+    """Compute one-hot featurization of string.
+
+    Parameters
+    ----------
+    string: str
+      An arbitrary string to be featurized.
 
     Returns
     -------
     np.ndarray
-      An one hot vector encoded from SMILES.
+      An one hot vector encoded from arbitrary input string.
       The shape is `(max_length, len(charset) + 1)`.
       The index of unknown character is `len(charset)`.
     """
-    try:
-      from rdkit import Chem
-    except ModuleNotFoundError:
-      raise ImportError("This class requires RDKit to be installed.")
-
-    smiles = Chem.MolToSmiles(mol)
     # validation
-    if len(smiles) > self.max_length:
+    if (len(string) > self.max_length):
       logger.info(
           "The length of {} is longer than `max_length`. So we return an empty array."
       )
       return np.array([])
 
-    smiles = self.pad_smile(smiles)
+    string = self.pad_string(string)  # Padding
     return np.array([
         one_hot_encode(val, self.charset, include_unknown_set=True)
-        for val in smiles
+        for val in string
     ])
 
+  def _featurize_mol(self, mol: RDKitMol) -> np.ndarray:
+    """Compute one-hot featurization of this molecule.
+
+    Parameters
+    ----------
+    mol: rdKit.Chem.rdchem.Mol
+      RDKit Mol object
+
+    Returns
+    -------
+    np.ndarray
+      An one hot vector encoded from SMILES.
+      The shape is '(max_length, len(charset) + 1)'
+      The index of unknown character is 'len(charset)'.
+    """
+    try:
+      from rdkit import Chem
+    except ModuleNotFoundError:
+      raise ImportError("This class requires RDKit to be installed.")
+    smiles = Chem.MolToSmiles(mol)  # Convert mol to SMILES string.
+    return self._featurize_string(smiles)  # Use string featurization.
+
   def pad_smile(self, smiles: str) -> str:
-    """Pad SMILES string to `self.pad_length`
+    """Pad smile string to `self.pad_length`
```

Comment:
It should be "SMILES", not "smile".  Same in the lines below.

Reply:
Good catch! Sorry about that.